runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -267,26 +267,39 @@ public ProjectionFunction getProjectionFunction() {
         }
     }
 
-    /** The "main" map which contains the primary key-value mappings. */
-    private final Map<K,V> mainMap = new HashMap<>();
+    // The "main" map which contains the primary key-value mappings.
+    private final Map<K,V> mainMap;
     private Set<Index<K, V, ? extends Comparable>> indexSpec = new HashSet<>();
     private final Map<String, Map<Comparable, Map<K, V>>> secondaryIndexes = new HashMap<>();
 
     @Getter
     boolean indexGenerationFailed = false;
 
-    /** Generate a table with the given set of indexes. */
-    public CorfuTable(IndexRegistry<K, V> indices) {
+    /**
+     * Generate a table with a given implementation for the mainMap
+     */
+    public CorfuTable(IndexRegistry<K, V> indices, Map<K, V> mapImpl) {
```

Comment:
parameter name `mapImpl` completle unclear

Reply:
I would refrain from changing CorfuTable in this PR and have that as a separate change. There are a lot more comments I am sure on CorfuTable than what you see here :)